File: backend/app/schemas/documents.py
from typing import Optional
from pydantic import BaseModel, ConfigDict, field_validator

from app.schemas.enums import DocumentProcessingTaskStatus, DocumentStatus


class DocumentCreate(BaseModel):
    title: str
    content: Optional[str] = None

    @field_validator("title")
    @classmethod
    def title_not_empty(cls, v: str) -> str:
        if not v.strip():
            raise ValueError("title cannot be empty")
        return v


class DocumentCreateResponse(BaseModel):
    celery_task_id: str
    document_id: int
    status: DocumentProcessingTaskStatus


class Document(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    status: DocumentStatus = DocumentStatus.DRAFT
    content: Optional[str] = None


type DocumentList = list[Document]

File: frontend/src/app/models/documents.ts
import { AppBaseResponse } from './system';

export enum DocumentStatus {
  DRAFT = 'draft',
  PUBLISHED = 'published',
  ARCHIVED = 'archived',
}

export enum DocumentProcessingTaskStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  COMPLETED = 'completed',
  FAILED = 'failed',
}

export interface Document {
  id: number;
  title: string;
  content?: string | null;
  status: DocumentStatus;
}

export type DocumentList = Document[];

export interface DocumentCreateRequest {
  title: string;
  content?: string | null;
}

export interface DocumentCreateResponseData {
  celery_task_id: string;
  document_id: number;
  status: DocumentProcessingTaskStatus;
}

export type DocumentListResponse = AppBaseResponse<DocumentList>;
export type DocumentDetailResponse = AppBaseResponse<Document>;
export type DocumentCreateResponse = AppBaseResponse<DocumentCreateResponseData>;
export type DocumentUpdateStatusResponse = AppBaseResponse<Document>;
